### PR TITLE
[KOGITO-3896] Link events has reverted multiple source/target behavior

### PR DESCRIPTION
**Pull request by fjtirado**
Carry forward of changes from JBPM to Kogito regarding links

### CODE REVIEW

File: jbpm/jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/ProcessHandler.java [hunk 1]
```diff
@@ -209,65 +216,91 @@ public Object end(final String uri, final String localName,
...
+            formatError(errors,"These nodes do not have a name ",noNameLinks.stream(), process);
+        }
+        if (!duplicatedTarget.isEmpty()) {
+            formatError(errors,"\nThere are multiple catch nodes with the same name ",duplicatedTarget.stream(), process);
+        }
+        if (!unconnectedTarget.isEmpty()) {
+            formatError(errors,"\nThere is not connection from any throw link to these catch links ",unconnectedTarget.stream(), process);
+        }
+        if (!throwLinks.isEmpty()) {
+            formatError(errors,"\nThere is not connection from any catch link to these throw links ",throwLinks
```
Comment (MarianMacik) [inline jbpm/jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/ProcessHandler.java:283]: Maybe a better message would be `There is not connection from these throw links to any catch link` or `There is not connection to any catch link from these throw links`. Just to satisfy the specification that the orientation of these links is from source/throw to target/catch.

File: jbpm/jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/ProcessHandler.java [hunk 2]
```diff
@@ -127,7 +132,8 @@ public ProcessHandler() {
 		}
 	}
 
-	public Object start(final String uri, final String localName,
+	@Override
+    public Object start(final String uri, final String localName,
```
Comment (cristianonicolai) [inline jbpm/jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/ProcessHandler.java:136]: unnecessary formatting change?

File: jbpm/jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/ProcessHandler.java [hunk 3]
```diff
@@ -183,7 +189,8 @@ public Object start(final String uri, final String localName,
 		return process;
 	}
 
-	@SuppressWarnings("unchecked")
+	@Override
+    @SuppressWarnings("unchecked")
```
Comment (cristianonicolai) [inline jbpm/jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/ProcessHandler.java:193]: unnecessary formatting change?

File: jbpm/jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/ProcessHandler.java [hunk 4]
```diff
@@ -303,7 +336,8 @@ private static Node findNodeByIdOrUniqueIdInMetadata(NodeContainer nodeContainer
 	     return node;
 	 }
 
-	 public Class<?> generateNodeFor() {
+	 @Override
+    public Class<?> generateNodeFor() {
```
Comment (cristianonicolai) [inline jbpm/jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/ProcessHandler.java:340]: unnecessary formatting change?

File: jbpm/jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/ProcessHandler.java (hunk 3 above)
Comment (fjtirado) [reply jbpm/jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/ProcessHandler.java:193]: Eclipse automatically adds the overrride and remove the old tabs, I think in this case is ok, since it is a pretty old file with old format

File: jbpm/jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/ProcessHandler.java (hunk 4 above)
Comment (fjtirado) [reply jbpm/jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/ProcessHandler.java:340]: Eclipse automatically adds the overrride and remove the old tabs, I think in this case is ok

File: jbpm/jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/ProcessHandler.java (hunk 1 above)
Comment (fjtirado) [reply jbpm/jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/ProcessHandler.java:283]: you are right, changed

File: jbpm/jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/ProcessHandler.java (hunk 2 above)
Comment (fjtirado) [reply jbpm/jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/ProcessHandler.java]: Keeping the override

**Comment by fjtirado** [comment]
jenkins retest this
@MarianMacik @cristianonicolai  Since monday and tueday is not working day, once green (it should be, last time failed because of a timeout but there were not real errors), please accept and merge, thank you!!

**Review by cristianonicolai — APPROVED** [review]

**Review by MarianMacik — APPROVED** [review]

**Comment by fjtirado** [comment]
@kiegroup/gatekeepers please merge this
